curves/bn254/src/lib.rs

```diff
@@ -247,31 +319,18 @@ mod target_arch {
         let ele_len = input.len().saturating_div(ALT_BN128_PAIRING_ELEMENT_LEN);
 
         let mut vec_pairs: Vec<(G1, G2)> = Vec::new();
```

Comment:
We know what size the vector is going to be, so we could already pre-allocate `ele_len` elements.

```suggestion
        let mut vec_pairs: Vec<(G1, G2)> = Vec::with_capacity(ele_len);
```